src/components/AuthContainer.js

```diff
@@ -16,7 +16,7 @@ import TermsOfService from 'src/pages/TermsOfService'
 const AuthContainer = ({ children }) => {
   const { name, public: isPublic } = useRoute()
   const { isSignedIn, registrationStatus, termsOfService, profile } = useStore(authStore)
-  const displayTosPage = isSignedIn && termsOfService.userCanUseTerra === false
+  const displayTosPage = isSignedIn && !termsOfService.permitsSystemUsage
```

Comment:
permitsSystemUsage can be undefined right, just want to double check there wont be an unintended codepath here.

Reply:
Yes, please check this. As is, this looks like it might be briefly showing the TOS page every time Terra is loaded. See https://github.com/DataBiosphere/terra-ui/pull/3678 for more information.